stko/utilities/utilities.py

```diff
@@ -1127,3 +1128,81 @@ def calculate_dihedral(pt1, pt2, pt3, pt4):
     x = np.dot(v, w)
     y = np.dot(np.cross(b1, v), w)
     return np.degrees(np.arctan2(y, x))
+
+
+def get_torsion_info_angles(mol: stk.ConstructedMolecule,
+                            torsion_info):
+    """gets the angles for torsion_info in mol
+
+    First angle returned is torsion angle in the
+    :class:`stk.ConstructedMolecule`
+    Second angle returned is torsion angle in the
+    :class:`stk.BuildingBlock`
+    Both angles are in degrees
+
+    A :class:`stko.MatchedTorsionCalculator` should yield torsions
+    such that the two angles returned are the same.
+    """
+    torsion = torsion_info.get_torsion()
+    angle = calculate_dihedral(
+        pt1=tuple(
+            mol.get_atomic_positions(
+                torsion.get_atom_ids()[0]
+            )
+        )[0],
+        pt2=tuple(
+            mol.get_atomic_positions(
+                torsion.get_atom_ids()[1]
+            )
+        )[0],
+        pt3=tuple(
+            mol.get_atomic_positions(
+                torsion.get_atom_ids()[2]
+            )
+        )[0],
+        pt4=tuple(
+            mol.get_atomic_positions(
+                torsion.get_atom_ids()[3]
+            )
+        )[0],
+    )
+    bb_torsion = torsion_info.get_building_block_torsion()
+    if bb_torsion is None:
+        bb_angle = None
+    else:
+        bb_angle = calculate_dihedral(
+            pt1=tuple(
+                torsion_info.get_building_block().get_atomic_positions(
+                    bb_torsion.get_atom_ids()[0]
+                )
+            )[0],
+            pt2=tuple(
+                torsion_info.get_building_block().get_atomic_positions(
+                    bb_torsion.get_atom_ids()[1]
+                )
+            )[0],
+            pt3=tuple(
+                torsion_info.get_building_block().get_atomic_positions(
+                    bb_torsion.get_atom_ids()[2]
+                )
+            )[0],
+            pt4=tuple(
+                torsion_info.get_building_block().get_atomic_positions(
+                    bb_torsion.get_atom_ids()[3]
+                )
+            )[0],
+        )
+    return angle, bb_angle
+
+
+def get_atom_maps(mol: stk.ConstructedMolecule):
+    """
+    map from building block atom ids to constructed molecule atoms
+    indexed by each building block id
+    """
+    atom_maps = defaultdict(dict)
+    for atom_info in mol.get_atom_infos():
+        current_atom_map = atom_maps[atom_info.get_building_block_id()]
+        bb_atom_id = atom_info.get_building_block_atom().get_id()
+        current_atom_map[bb_atom_id] = atom_info.get_atom()
```

Comment:
I might be a fool here, but when does `atom_maps` get updated? are you relying on the pointer nature of that object?

Reply:
I see - I am sure that works but it makes this code hard to read. Can you do the update to atom_maps explicitly?

```python
atom_maps[atom_info.get_building_block_id()][bb_atom_id] = atom_info.get_atom()
```

Reply:
I can, just not sure how to break that line up in a readable way.